common/beerocks/bcl/unit_tests/socket_event_loop_test.cpp

```diff
@@ -0,0 +1,189 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <bcl/beerocks_socket_event_loop.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <easylogging++.h>
+
+// Initialize easylogging++
+INITIALIZE_EASYLOGGINGPP
+
+using namespace beerocks;
+
+using ::testing::InSequence;
+using ::testing::Invoke;
+using ::testing::StrictMock;
+
+static constexpr std::chrono::milliseconds s_test_timeout(100);
+
+/**
+ * @brief Mockable event handlers class
+ */
+class EventHandlersMock : public EventLoop::EventHandlers {
+
+public:
+    EventHandlersMock()
+    {
+        on_read = [&](std::shared_ptr<Socket> socket, EventLoop &loop) {
+            return handle_read(socket, &loop);
+        };
+        on_write = [&](std::shared_ptr<Socket> socket, EventLoop &loop) {
+            return handle_write(socket, &loop);
+        };
+        on_timeout = [&](std::shared_ptr<Socket> socket, EventLoop &loop) {
+            return handle_timeout(socket, &loop);
+        };
+        on_disconnect = [&](std::shared_ptr<Socket> socket, EventLoop &loop) {
+            return handle_disconnect(socket, &loop);
+        };
+        on_error = [&](std::shared_ptr<Socket> socket, EventLoop &loop) {
+            return handle_error(socket, &loop);
+        };
+    }
+
+    MOCK_METHOD(bool, handle_read, ((std::shared_ptr<Socket> &), EventLoop *));
+    MOCK_METHOD(bool, handle_write, ((std::shared_ptr<Socket> &), EventLoop *));
+    MOCK_METHOD(bool, handle_timeout, ((std::shared_ptr<Socket> &), EventLoop *));
+    MOCK_METHOD(bool, handle_disconnect, ((std::shared_ptr<Socket> &), EventLoop *));
+    MOCK_METHOD(bool, handle_error, ((std::shared_ptr<Socket> &), EventLoop *));
+};
+
+TEST(beerocks_socket_event_loop, setup)
```

Comment:
Test order is undefined, so your code can't depend on a test preceding or following another.
If you want some code to be executed before each test, then consider using a [text fixture](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests)

Reply:
Interesting.. So far it seems like the tests are always executed in the order they are defined in the file.

Reply:
You can run the executable with command-line options to run only one specific (set of) tests.